### PR TITLE
#392: Skip parser rebuilding, use configuration file, and fix benchmarks

### PR DESCRIPTION
**Pull request by jarulraj**
This PR:

- Skips rebuilding parser every single time
- Moves the configuration folder to the root directory
- Expands the Google flags configuration file
- Fixes compilation of benchmarks

Let's start using the configuration file for setting runtime parameters.

### CODE REVIEW

File: configuration/peloton.conf [hunk 1]
```diff
@@ -0,0 +1,119 @@
+# -----------------------------
```
Comment (malin1993ml) [inline configuration/peloton.conf]: Where did you load this file for the server? I just didn't find the code... Did you specify it somewhere in the CMake file?

File: configuration/peloton.conf [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+#                GB = gigabytes                     s   = seconds
+#                                                   min = minutes
+#                                                   h   = hours
+
+#------------------------------------------------------------------------------
+# FILE LOCATIONS
+#------------------------------------------------------------------------------
+
+# Location of data directory
+--data_directory='data'
```
Comment (malin1993ml) [inline configuration/peloton.conf]: For parameter like this, how the system can accept it as a command-line argument? I was using GFLAG to parser and accept arguments like "stats_mode" or "port" from command line. But I didn't see the GFLAG declaration of "data_directory".

File: configuration/peloton.conf [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+--socket_family=AF_INET
+
+#------------------------------------------------------------------------------
+# RESOURCE USAGE
+#------------------------------------------------------------------------------
+
+# - Memory -
+
+# Shared memory pool
+--shared_pool=128MB
```
Comment (malin1993ml) [inline configuration/peloton.conf]: Same question here.

File: configuration/peloton.conf (hunk 1 above)
Comment (jarulraj) [inline configuration/peloton.conf]: It is loaded at runtime like this:

`./bin/peloton -flagfile ../configuration/peloton.conf`

File: configuration/peloton.conf (hunk 2 above)
Comment (jarulraj) [inline configuration/peloton.conf]: I have not yet added some of those parameters or used them inside peloton. 

But, they should eventually be declared in `src/include/configuration/configuration.h` and defined in `src/configuration/configuration.cpp`. They can then be used in any module within peloton like a regular GFLAGs parameter.

Comment (malin1993ml) [inline configuration/peloton.conf]: Okay.

**Comment by malin1993ml** [comment]
You may have to rebase now since there're some conflicts with the latest merged PR.

**Comment by jarulraj** [comment]
Sure, I have rebased it.

**Comment by jarulraj** [comment]
I am not sure why the jdbc validator failed.

**Pull request merged**